python/ray/tune/registry.py

```diff
@@ -33,19 +34,18 @@ def register_trainable(name, trainable):
 
     from ray.tune.trainable import Trainable, wrap_function
 
-    if isinstance(trainable, type):
-        logger.debug("Detected class for trainable.")
-    elif isinstance(trainable, FunctionType):
-        logger.debug("Detected function for trainable.")
-        trainable = wrap_function(trainable)
+    if isinstance(trainable, six.class_types) and issubclass(trainable, Trainable):
+        logger.debug(
+            "Detected `Trainable` subclass for trainable. Using the class API")
     elif callable(trainable):
-        logger.warning(
-            "Detected unknown callable for trainable. Converting to class.")
+        logger.debug("Detected callable for trainable. Using the function API.")
```

Comment:
I think this automatically wraps classes that are not Trainables?

Reply:
I was under the impression that this was the desired behavior. I might be misunderstanding something from our last exchange. Why should `functools.partial`'ed functions be treated differently or, more generally, what is the purpose of distinguishing `callable` classes from "pure" functions?

I don't think this would lead to any unexpected behavior since, currently, it is already supported but with logs a warning. Is this warning necessary, and, if so, what kind of user error or trap is it trying to avoid?